### PR TITLE
OHRM5X-1278:  forgot password bugs

### PR DESCRIPTION
**Pull request by Ajiharan007**

### CODE REVIEW

File: src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php [hunk 1]
```diff
@@ -186,13 +191,20 @@ protected function generatePasswordResetEmailBody(Employee $receiver, string $re
...
+            $this->getEmailService()->setMessageTo([$receiver->getWorkEmail()]);
+            $this->getEmailService()->setMessageFrom(
+                [$this->getEmailService()->getEmailConfig()->getSentAs() => 'OrangeHRM']
+            );
+            $this->getEmailService()->setMessageSubject('OrangeHRM Password Reset');
+            $this->getEmailService()->setMessageBody(
+                $this->generatePasswordResetEmailBody($receiver, $resetCode, $userName)
+            );
+            return $this->getEmailService()->sendEmail();
+        } catch (TransportExceptionInterface $e) {
```
Comment (Ajiharan007) [inline src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php:204]: catch whether send email  with invalid email configuration

File: src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php [hunk 2]
```diff
@@ -112,8 +113,7 @@ public function generateEmailBody(string $templateFile, array $placeholders, arr
             ) . '/orangehrmAuthenticationPlugin/config/data' . '/' . $templateFile
         );
 
-        $body = str_replace($placeholders, $replacements, $body);
-        return $body;
+        return nl2br(str_replace($placeholders, $replacements, $body));
```
Comment (Ajiharan007) [inline src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php:116]: add new line for email template

File: src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php [hunk 3]
```diff
@@ -112,8 +112,7 @@ public function generateEmailBody(string $templateFile, array $placeholders, arr
             ) . '/orangehrmAuthenticationPlugin/config/data' . '/' . $templateFile
```
Comment (RajithaKumara) [inline src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php:113]:
```suggestion
            ) . '/orangehrmAuthenticationPlugin/config/data/' . $templateFile
```
:thinking:

File: src/plugins/orangehrmAuthenticationPlugin/test/fixtures/ResetPasswordService.yml [hunk 4]
```diff
@@ -61,9 +62,16 @@ UserRole:
...
+  5: { id: 6, user_role_id: 1, emp_number: 7, user_name: 'Admin', user_password: 'Renukshan', status: 1, date_entered: '2011-04-12', date_modified: '2011-04-13' }
 
 ResetPasswordRequest:
   0: {id: 1, reset_email: 'haran@orangehrm.live.com', reset_request_date: '2022-04-01 06:59:35', reset_code: 'YWRtaW4jU0VQQVJBVE9SI-xpEY5IF4lNPp8bfWQzz2Q'}
   1: {id: 2, reset_email: 'haran@orangehrm.live.com', reset_request_date: '2022-04-05 09:59:35', reset_code: 'YWRtaW4jU0VQQVJBVE9SI42zK2G1KccR7cvAP31Dgnw'}
+  3: {id: 3, reset_email: 'renukshan@xample.com', reset_request_date: '2022-04-05 09:55:35', reset_code: 'YWRtaW4jU0VQQVJBVE9SI42zK2G1KccR7cvAP31Dgnw'}
+  4: {id: 4, reset_email: 'Ajiharan@orangehrmlive.com', reset_request_date: '2022-04-05 09:55:35', reset_code: 'QWRtaW4jU0VQQVJBVE9SI02O9C_ScxCx_t5U1tKybS0'}
+
+EmailConfiguration:
+  0: {id: 1, mail_type: 'smtp', sent_as: 'test@orangehrm.com', smtp_host: 'smtp.mailtrap.io', smtp_port: 587, smtp_username: 'test@orangehrm.com', smtp_password: '0000', smtp_auth_type: 'login', smtp_security_type: 'tls'}
```
Comment (RajithaKumara) [inline src/plugins/orangehrmAuthenticationPlugin/test/fixtures/ResetPasswordService.yml:77]:
```suggestion
  0: {id: 1, mail_type: 'smtp', sent_as: 'test@orangehrm.com', smtp_host: 'smtp.example.com', smtp_port: 587, smtp_username: 'test@orangehrm.com', smtp_password: '0000', smtp_auth_type: 'login', smtp_security_type: 'tls'}
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php [hunk 5]
```diff
@@ -142,6 +142,11 @@ public function searchForUserRecord(string $username): ?User
             return null;
         }
 
+        if (!$user->getStatus()) {
+            $this->getLogger()->error('user Account was disabled');
```
Comment (RajithaKumara) [inline src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php:146]:
```suggestion
            $this->getLogger()->error('Account disabled');
```

File: src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php (hunk 1 above)
Comment (RajithaKumara) [reply src/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php:204]: :ok_hand:

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**